lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -815,13 +826,33 @@ static void optimizeBatchNorm(Function *F) {
       // Q = W * A
       // C = b * A + B
 
-      auto filterH = cast<Variable>(CV->getFilter())->getHandle<>();
-      auto cbiasH = cast<Variable>(CV->getBias())->getHandle<>();
+      // FIXME: We need to use a temporary node to hold on
+      // the variables, because using the result directly
+      // will create a temporary NodeValue that is going
+      // to be otherwise alive at the call site and that
+      // messes up with the number of users.
+      Node *tmp = CV->getFilter().getNode();
```

Comment:
I wonder why getFilter returns NodeValue instead of NodeValue&. But that's a different story.

Reply:
Agreed :).